### PR TITLE
Introduce rule accounts_passwords_pam_faillock_audit

### PR DESCRIPTION
**Pull request by Xeicker**
#### Description:

- Introduce rule `accounts_passwords_pam_faillock_audit`
- Added OVAL, Bash, Ansible and tests  to rule 

#### Rationale:

- This covers DISA STIG Ids `OL08-00-020020` & `OL08-00-020021`

### CODE REVIEW

File: linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/account_passwords_pam_faillock_audit/oval/shared.xml [hunk 1]
```diff
@@ -0,0 +1,111 @@
+<def-group>
+    <definition class="compliance" id="{{{ rule_id }}}" version="5">
+        {{{
+        oval_metadata("Prevent System Messages When Three Unsuccessful Logon Attempts Occur")
```
Comment (jan-cerny) [inline linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/account_passwords_pam_faillock_audit/oval/shared.xml:4]: The title is wrong

File: linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/account_passwords_pam_faillock_audit/ansible/shared.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# reboot = false
+# strategy = restrict
+# complexity = low
+# disruption = low
+{{{ ansible_pam_faillock_enable() }}}
+
+- name: {{{ rule_title }}} - Check the presence of /etc/security/faillock.conf file
+  ansible.builtin.stat:
+    path: /etc/security/faillock.conf
+  register: result_faillock_conf_check
```
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/account_passwords_pam_faillock_audit/ansible/shared.yml:11]: Why not using the `ansible_pam_faillock_parameter_value` macro? So we keep all the pam_faillock rules aligned and avoid duplicated code.

File: linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/account_passwords_pam_faillock_audit/bash/shared.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
+# platform = multi_platform_rhel,multi_platform_fedora,multi_platform_ol,multi_platform_rhv,multi_platform_sle
+
+{{{ bash_pam_faillock_enable() }}}
+AUTH_FILES=("/etc/pam.d/system-auth" "/etc/pam.d/password-auth")
```
Comment (marcusburghardt) [inline linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/account_passwords_pam_faillock_audit/bash/shared.sh:4]: Like in the comment in `shared.yml`, I recommend to use the `bash_pam_faillock_parameter_value` macro.

**Review by jan-cerny — APPROVED** [review]
thanks

**Pull request merged**